src/intrinsics/MapPrototype.mts

```diff
@@ -118,6 +118,75 @@ function MapProto_get([key = Value.undefined]: Arguments, { thisValue }: Functio
   return Value.undefined;
 }
 
+/**  https://tc39.es/proposal-upsert/#sec-map.prototype.getOrInsert */
+function MapProto_getOrInsert([key = Value.undefined, value = Value.undefined]: Arguments, { thisValue }: FunctionCallContext): ValueCompletion {
+  // 1. Let M be the this value.
+  const M = thisValue as MapObject;
+  // 2. Perform ? RequireInternalSlot(M, [[MapData]]).
+  Q(RequireInternalSlot(M, 'MapData'));
+  // 3. Set key to CanonicalizeKeyedCollectionKey(key).
+  if (key instanceof NumberValue && Object.is(R(key), -0)) {
+    key = F(+0);
+  }
+  // 4. For each Record { [[Key]], [[Value]] } p of M.[[MapData]], do
+  const entries = M.MapData;
+  for (const p of entries) {
+    // a. If p.[[Key]] is not empty and SameValue(p.[[Key]], key) is true, return p.[[Value]].
+    if (p.Key !== undefined && SameValueZero(p.Key, key) === Value.true) {
```

Comment:
Is it SV or SVZ? Comment and the code are conflict

Reply:
It's SV, thanks for catching that.